blockchain/upgrade.go

```diff
@@ -1009,5 +1387,14 @@ func upgradeDB(ctx context.Context, db database.DB, chainParams *chaincfg.Params
 		}
 	}
 
+	// Update to a version 7 database if needed.  This entails migrating the
+	// block index to the version 3 format.
+	if dbInfo.version == 6 && dbInfo.bidxVer < 3 {
```

Comment:
Previously, the software didn't complain about newer versions other than the database version, so it could lead to old code trying to run against an updated index.  However, now that everyone is required to be running a version which will error in that circumstance, it seems fair to limit this specifically to `bidxVer == 2`.

I'll get it updated over the next couple of days.

Reply:
Updated.